### ISSUE TITLE
Button: Fix `SUCCESS` tone color contrast

### ISSUE THREAD

**Issue by fk** [gatsby-inc/gatsby-interface#305]
`<Button tone="SUCCESS">` currently fails the WCAG 2 AA color contrast ratio:

![image](https://user-images.githubusercontent.com/21834/80284123-c0b2d180-871c-11ea-8b4b-0c7f32e76723.png)

Let's fix this analog to what we're already doing for `<Notification tone="SUCCESS">` and `<Badge tone="SUCCESS">`, and get rid of the following conditional styles by adjusting the "SUCCESS" `tone` accordingly:

- https://github.com/gatsby-inc/gatsby-interface/blob/665648fb8ae0a8aeaffda912dcaa47b2bc1a05ab/src/components/Badge/Badge.helpers.tsx#L29-L31
- https://github.com/gatsby-inc/gatsby-interface/blob/665648fb8ae0a8aeaffda912dcaa47b2bc1a05ab/src/components/Badge/Badge.helpers.tsx#L14-L16
- https://github.com/gatsby-inc/gatsby-interface/blob/665648fb8ae0a8aeaffda912dcaa47b2bc1a05ab/src/components/Notification/Notification.helpers.ts#L55-L57
